gravitee-am-identityprovider/gravitee-am-identityprovider-oauth2-generic/src/test/java/io/gravitee/am/identityprovider/oauth2/authentication/OAuth2GenericAuthenticationProviderTest_idToken.java

```diff
@@ -73,7 +76,44 @@ public Object getPrincipal() {
             public AuthenticationContext getContext() {
                 DummyRequest dummyRequest = new DummyRequest();
                 dummyRequest.setParameters(Collections.singletonMap("urlHash", Collections.singletonList("#id_token=test")));
-                return new DummyAuthenticationContext(Collections.singletonMap("id_token", "test"), dummyRequest);
+                return new DummyAuthenticationContext(new HashMap<>(Collections.singletonMap("id_token", "test")), dummyRequest);
+            }
+        }).test();
+
+        testObserver.assertComplete();
+        testObserver.assertNoErrors();
+        testObserver.assertValue(u -> "bob".equals(u.getUsername()));
+    }
+
+    @Test
+    public void shouldLoadUserByUsername_authentication_and_filter_undesired_claims() throws ParseException, JOSEException, BadJOSEException {
+        JWTClaimsSet claims = new JWTClaimsSet.Builder().subject("bob")
+            .claim(Claims.nbf, "value that won't appear")
+            .claim(Claims.iat, "value that won't appear")
+            .claim(Claims.exp, "value that won't appear")
+            .claim(ConstantKeys.OIDC_PROVIDER_ID_TOKEN_KEY, "value that won't appear")
+            .claim(ConstantKeys.OIDC_PROVIDER_ID_ACCESS_TOKEN_KEY, "value that won't appear")
```

Comment:
I can't see the assertions that check that these claims are missing :monocle_face: